README.md

```diff
@@ -237,6 +237,113 @@ export KUBERNETES_CONFORMANCE_TEST=y
 go run hack/e2e.go -v --test --test_args="--ginkgo.focus=\[Conformance\]" --check_version_skew=false --check_node_count=false
 ```
 
+#### Adding a New Dependency
+
+Minikube uses `Godep` to manage vendored dependencies.
+`Godep` can be a bit finnicky with a project with this many dependencies.
+Here is a rough set of steps that usually works to add a new dependency.
+
+1. Make a clean GOPATH, with minikube in it.
+This isn't strictly necessary, but it usually helps.
+
+```shell
+mkdir -p $HOME/newgopath/src/k8s.io
+export GOPATH=$HOME/newgopath
+cd $HOME/newgopath/src/k8s.io
+git clone https://github.com/kubernetes/minikube.git
+```
+
+2. `go get` your new dependency.
+```shell
+go get mynewdepenency
+```
+
+3. Use it in code, build and test.
+
+4. Import the dependency from GOPATH into vendor/
+```shell
+godep save ./...
+```
+
+If it is a large dependency, please commit the vendor/ directory changes separately.
+This makes review easier in Github.
+
+```shell
+git add vendor/
+git commit -m "Adding dependency foo"
+git add --all
+git commit -m "Adding cool feature"
+```
+
+#### Updating Kubernetes
+
+To update Kubernetes, follow these steps:
+
+1. Make a clean GOPATH, with minikube in it.
+This isn't strictly necessary, but it usually helps.
+
+ ```shell
+ mkdir -p $HOME/newgopath/src/k8s.io
+ export GOPATH=$HOME/newgopath
+ cd $HOME/newgopath/src/k8s.io
+ git clone https://github.com/kubernetes/minikube.git
+ ```
+
+2. Copy your vendor directory back out to the new GOPATH.
+
+ ```shell
+ cd minikube
+ godep restore ./...
+ ```
+
+3. Kubernetes should now be on your GOPATH. Check it out to the right version.
+Make sure to also fetch tags, as Godep relies on these.
+
+ ```shell
+ cd $GOPATH/src/k8s.io/kubernetes
+ git fetch --tags
+ ```
+ 
+ Then list all available Kubernetes tags:
+
+ ```shell
+ git tag
+ ...
+ v1.2.4
+ v1.2.4-beta.0
+ v1.3.0-alpha.3
+ v1.3.0-alpha.4
+ v1.3.0-alpha.5
+ ...
+```
+
+ Then checkout the correct one and update its dependencies with:
+ 
+ ```shell
+ git checkout $DESIREDTAG
+ godep restore ./...
+ ```
+
+4. Build and test minikube, making any manual changes necessary to build.
+
+5. Update godeps
+
+ ```shell
+ cd $GOPATH/src/k8s.io/minikube
+ rm -rf Godeps/ vendor/
+ godep save ./...
+ ```
+
```

Comment:
Perhaps add a step to verify that the Kubernetes Version was changed:
// Verify that the correct Kubernetes version is being used by minikube:
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.4", GitCommit:"3eed1e3be6848b877ff80a93da3785d9034d0a4f", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"3+", GitVersion:"v1.3.0-beta.2", GitCommit:"caf9a4d87700ba034a7b39cced19bd5628ca6aa3", GitTreeState:"clean"}